fsspec_reference_maker/combine.py

```diff
@@ -316,3 +278,53 @@ def _determine_dims(self):
         )
         self.same_dims = set(ds.dims) - self.extra_dims - self.concat_dims
         return ds, ds0, fss
+
+
+def make_coord(fss, z, accum_dim):
+    # a)
+    accum = []
+    logger.debug("accumulate coords array %s", accum_dim)
+    times = False
+    for fs in fss:
+        zz = zarr.open_array(fs.get_mapper(accum_dim))
+
+        try:
+            import cftime
+            if not isinstance(zz, cftime.real_datetime):
+
+                # Try and get the calendar attribute from "calendar" attribute
+                # If it doesn't exist, assume a standard calendar
+                if zz.attrs.get("calendar") is not None:
+                    calendar = zz.attrs.get("calendar")
+                else:
+                    calendar = 'standard'
+
+                    # Update attrs in z[accum_dim]
+                    zattr = dict(z[accum_dim].attrs)
+                    zattr['calendar'] = 'standard'
+                    z[accum_dim].attrs.put(zattr)            
+                
+                zz = cftime.num2pydate(zz[...], units=zz.attrs["units"],
+                                       calendar=calendar)
+                times = True
+                logger.debug("converted times")
+                accum.append(zz)
+            else:
+                accum.append(zz)
+        except Exception as e:
+            ex = e
+            accum.append(zz[...].copy())
+    attr = dict(z[accum_dim].attrs)
+    if times:
+        accum = [np.array(a, dtype="M8") for a in accum]
+        attr.pop('units')
+        attr.pop('calendar')
```

Comment:
Didn't you say that we should remove "calendar" in every case, since we didn't necessary get to `times = True` after setting it in the block above?

Reply:
Sure, but `.pop()` raises `KeyError` if there's no "calendar" attribute. So maybe something like this

```python
for key in ('units', 'calendar'):
  try:
    attr.pop(key)
  except KeyError:
    pass

Reply:
`pop("calendar", None)`